### PR TITLE
Add a separate queue for querying PACS

### PR DESCRIPTION
**Pull request by p-j-smith**
## Description
Fixes https://github.com/SAFEHR-data/the-rolling-skeleton/issues/416

- add a second queue (`_imaging_secondary`) for querying PACS
- messages processed by the existing `imaging` queue will only query the VNA. If a study is not available in the VNA, the corresponding message will be published to the new secondary imaging queue
- messages are only processed by the secondary imaging queue outside of working hours (as we don't want to query PACS during working hours)

## Type of change
Please delete options accordingly to the description.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### Suggested Checklist

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have passed on my local host device. (see further details at the [CONTRIBUTING](https://github.com/SAFEHR-data/PIXL/blob/main/CONTRIBUTING.md#local-testing) document)
- [ ] Make sure your branch is up-to-date with main branch. See [CONTRIBUTING](https://github.com/SAFEHR-data/PIXL/blob/main/CONTRIBUTING.md) for a general example to syncronise your branch with the `main` branch.
- [x] I have requested PR review to `UCLH-Foundtry/arc-dev`
- [x] I have adddressed and marked as resolved all the review comments in my PR.
- [ ] Finally, I have selected `squash and merge`

### CODE REVIEW

**Review by stefpiatek — APPROVED** [review]
Some of this may be outdated because I thought I submitted 🤦🏻

**Comment by stefpiatek** [comment]
Oh also its worth checking the checklist in https://github.com/SAFEHR-data/the-rolling-skeleton/issues/416 to make sure that we've covered all of that

File: cli/src/pixl_cli/_config.py [hunk 1]
```diff
@@ -68,7 +68,11 @@ def base_url(self) -> str:
 
 def api_config_for_queue(queue_name: str) -> APIConfig:
     """Configuration for an API associated with a queue"""
-    api_name = f"{queue_name}_api"
+    api_name = {
+        "export": "export_api",
+        "imaging-primary": "imaging_api",
+        "imaging-secondary": "imaging_api",
```
Comment (stefpiatek) [inline cli/src/pixl_cli/_config.py:74]: oooh this is nice

File: cli/src/pixl_cli/_message_processing.py [hunk 2]
```diff
@@ -120,10 +120,17 @@ def _wait_for_queues_to_empty(queues_to_populate: list[str]) -> None:
 
 
 def _message_count(queues_to_populate: list[str]) -> int:
+    # We don't want to modify the queues we're populating, but if we're populating imaging-primary
+    # we also need to wait for imaging-secondary to be empty
+    queues_to_count = queues_to_populate.copy()
+    if "imaging-primary" in queues_to_populate and "imaging-secondary" not in queues_to_populate:
+        queues_to_count.append("imaging-secondary")
```
Comment (stefpiatek) [inline cli/src/pixl_cli/_message_processing.py:127]: could make just make this a set and then add it to it without checking?

**Review by stefpiatek — APPROVED** [review]
Looking good, would be nice to do set off a run overnight tonight to check that it successfuly runs through both queues still (no need to take volumes down)

**Comment by p-j-smith** [comment]
> Looking good, would be nice to do set off a run overnight tonight to check that it successfuly runs through both queues still (no need to take volumes down)

yeah good shout - I've set it running on `pixl_test`

**Comment by p-j-smith** [comment]
Patch coverage is a bit low but it's mostly on hard to test areas (we're mocking a lot of these parts of the code during tests), so hopefully it's okay to merge as is

**Pull request merged**